packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -1,13 +1,19 @@
 import * as childProcess from 'child_process';
+import * as os from 'os';
 import * as path from 'path';
 import * as cxschema from '@aws-cdk/cloud-assembly-schema';
 import * as cxapi from '@aws-cdk/cx-api';
 import * as fs from 'fs-extra';
-import { debug } from '../../logging';
+import * as semver from 'semver';
+import { debug, warning } from '../../logging';
 import { Configuration, PROJECT_CONFIG, USER_DEFAULTS } from '../../settings';
+import { loadTree, some } from '../../tree';
+import { splitBySize } from '../../util/objects';
 import { versionNumber } from '../../version';
 import { SdkProvider } from '../aws-auth';
 
+const ENV_VARIABLE_SIZE_LIMIT = 131072; // 128KiB
```

Comment:
> Linux systems don't support environment variables larger than 128KiB.

We also have to think about Windows. In Windows, the maximum size of the WHOLE environment together is 32KiB. We can do some calculation based on the current environment to come up with a max for the context value, or do a rough estimation and say we'll cap it at 16KiB or something.

Reply:
Good call out!